File: LeniorClient.java
// LeniorClient.java
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Duration;
import java.util.Base64;
import java.util.Scanner;

public class LeniorClient {

    private static final String API_BASE = "https://lenior-api-1-hvvj.onrender.com";
    private static final String TEXT_ENDPOINT = API_BASE + "/chat/texto";
    private static String sessaoId = null;

    public static void main(String[] args) throws Exception {
        HttpClient client = HttpClient.newBuilder()
                .connectTimeout(Duration.ofSeconds(10))
                .build();

        System.out.println("🧠 Lenior 2.0 — Cliente Java");
        System.out.println("Digite 'sair' para encerrar.\n");

        try (Scanner scanner = new Scanner(System.in)) {
            while (true) {
                System.out.print("Você: ");
                String pergunta = scanner.nextLine().trim();
                if (pergunta.isEmpty()) continue;
                if (pergunta.equalsIgnoreCase("sair") || pergunta.equalsIgnoreCase("quit")) {
                    System.out.println("Até logo!");
                    break;
                }

                String jsonPayload = String.format("{\"texto\": \"%s\"%s}",
                        pergunta.replace("\"", "\\\""),
                        sessaoId != null ? ", \"sessao_id\": \"" + sessaoId + "\"" : "");

                HttpRequest request = HttpRequest.newBuilder()
                        .uri(URI.create(TEXT_ENDPOINT))
                        .header("Content-Type", "application/json")
                        .POST(HttpRequest.BodyPublishers.ofString(jsonPayload))
                        .timeout(Duration.ofSeconds(30))
                        .build();

                try {
                    HttpResponse<String> response = client.send(request,
                            HttpResponse.BodyHandlers.ofString());

                    if (response.statusCode() != 200) {
                        System.out.println("❌ Erro HTTP " + response.statusCode());
                        System.out.println(response.body());
                        continue;
                    }

                    String body = response.body();
                    String texto = extrairCampo(body, "\"texto\"");
                    String audioBase64 = extrairCampo(body, "\"audio\"");
                    String novaSessao = extrairCampo(body, "\"sessao_id\"");

                    if (novaSessao != null && !novaSessao.isEmpty()) {
                        sessaoId = novaSessao;
                    }

                    System.out.println("Lenior: " + (texto != null ? texto : "(vazio)"));

                    if (audioBase64 != null && !audioBase64.isEmpty()) {
                        byte[] audioBytes = Base64.getDecoder().decode(audioBase64);
                        String nomeArq = "resposta_java.wav";
                        java.nio.file.Files.write(java.nio.file.Paths.get(nomeArq), audioBytes);
                        System.out.println("✅ Áudio salvo em: " + nomeArq);
                    }

                    if (body.contains("\"execucao\"")) {
                        String exec = extrairCampo(body, "\"execucao\"");
                        System.out.println("🧪 Execução: " + (exec != null ? exec : "(sem detalhes)"));
                    }

                } catch (Exception e) {
                    System.err.println("❌ Erro: " + e.getMessage());
                }
                System.out.println();
            }
        }
    }

    private static String extrairCampo(String json, String chave) {
        String busca = "\"" + chave + "\":";
        int idx = json.indexOf(busca);
        if (idx < 0) return null;
        int ini = idx + busca.length();
        while (ini < json.length() && Character.isWhitespace(json.charAt(ini))) ini++;
        if (ini >= json.length()) return null;
        char primeiro = json.charAt(ini);
        if (primeiro == '"') {
            int fim = ini + 1;
            while (fim < json.length() && json.charAt(fim) != '"') fim++;
            if (fim < json.length()) {
                return json.substring(ini + 1, fim);
            }
            return null;
        } else if (primeiro == '{' || primeiro == '[') {
            int profundidade = 0;
            int fim = ini;
            do {
                char c = json.charAt(fim);
                if (c == '{' || c == '[') profundidade++;
                else if (c == '}' || c == ']') profundidade--;
                fim++;
            } while (fim < json.length() && profundidade > 0);
            return json.substring(ini, fim);
        } else {
            int fim = ini;
            while (fim < json.length() && ",}] \t\n\r".indexOf(json.charAt(fim)) < 0) fim++;
            return json.substring(ini, fim);
        }
    }
}

File: lenior_client.py
#!/usr/bin/env python3
"""
lenior_client.py — Cliente Python para a API Lenior 2.0
Uso: python lenior_client.py "sua pergunta aqui"
"""

import requests
import json
import sys
import os
import base64
from pathlib import Path

# ===== CONFIGURAÇÃO =====
API_BASE = "https://lenior-api-1-hvvj.onrender.com"
TEXT_ENDPOINT = f"{API_BASE}/chat/texto"
AUDIO_ENDPOINT = f"{API_BASE}/chat/audio"

# ===== SESSÃO (persistente) =====
SESSION_FILE = Path.home() / ".lenior_session"

def carregar_sessao():
    if SESSION_FILE.exists():
        with open(SESSION_FILE, "r") as f:
            return f.read().strip()
    return None

def salvar_sessao(sessao_id):
    with open(SESSION_FILE, "w") as f:
        f.write(sessao_id)

# ===== FUNÇÕES =====

def perguntar_texto(pergunta, sessao_id=None):
    """Envia uma pergunta de texto e retorna a resposta."""
    payload = {"texto": pergunta}
    if sessao_id:
        payload["sessao_id"] = sessao_id

    resp = requests.post(TEXT_ENDPOINT, json=payload)
    resp.raise_for_status()
    return resp.json()

def perguntar_audio(caminho_audio, sessao_id=None):
    """Envia um arquivo de áudio e retorna a resposta."""
    with open(caminho_audio, "rb") as f:
        files = {"audio": (os.path.basename(caminho_audio), f, "audio/mp3")}
        data = {}
        if sessao_id:
            data["sessao_id"] = sessao_id
        resp = requests.post(AUDIO_ENDPOINT, files=files, data=data)
        resp.raise_for_status()
        return resp.json()

def salvar_audio_base64(base64_str, nome_saida="resposta_audio.wav"):
    """Salva o áudio retornado em base64 para um arquivo."""
    if not base64_str:
        print("⚠️ Nenhum áudio retornado.")
        return
    audio_bytes = base64.b64decode(base64_str)
    with open(nome_saida, "wb") as f:
        f.write(audio_bytes)
    print(f"✅ Áudio salvo em: {nome_saida}")

# ===== MAIN =====

def main():
    sessao_id = carregar_sessao()

    if len(sys.argv) > 1:
        pergunta = " ".join(sys.argv[1:])
        print(f"🧠 Você: {pergunta}")
        try:
            data = perguntar_texto(pergunta, sessao_id)
            if data.get("sessao_id"):
                salvar_sessao(data["sessao_id"])
            print(f"🤖 Lenior: {data.get('texto', '(vazio)')}")
            if data.get("audio"):
                salvar_audio_base64(data["audio"])
            if data.get("execucao"):
                print(f"🧪 Execução: {json.dumps(data['execucao'], indent=2, ensure_ascii=False)}")
        except Exception as e:
            print(f"❌ Erro: {e}")
        return

    # Modo interativo
    print("🧠 Lenior 2.0 — Cliente Python")
    print("Digite 'sair' para encerrar.\n")
    while True:
        try:
            pergunta = input("Você: ").strip()
        except (EOFError, KeyboardInterrupt):
            print("\nAté logo!")
            break
        if pergunta.lower() in ("sair", "quit", "exit"):
            print("Até logo!")
            break
        if not pergunta:
            continue

        try:
            data = perguntar_texto(pergunta, sessao_id)
            if data.get("sessao_id"):
                salvar_sessao(data["sessao_id"])
            print(f"Lenior: {data.get('texto', '(vazio)')}")
            if data.get("audio"):
                salvar_audio_base64(data["audio"])
            if data.get("execucao"):
                print(f"🧪 Execução: {json.dumps(data['execucao'], indent=2, ensure_ascii=False)}")
            print()
        except Exception as e:
            print(f"❌ Erro: {e}\n")

if __name__ == "__main__":
    main()
